spec/models/import/transactions/actual_and_refund_csv_row_spec.rb

```diff
@@ -0,0 +1,583 @@
+require "rails_helper"
+
+RSpec.describe Import::Transactions::ActualAndRefundCsvRow, type: :model do
+  before do
+    @providing_organisation = build(:partner_organisation)
+    @activity = build(:project_activity, title: "Activity Title", organisation: @providing_organisation)
+    allow(@activity).to receive(:providing_organisation).and_return(@providing_organisation)
+    allow(Activity).to receive(:find_by_roda_identifier).and_return(@activity)
+  end
+
+  describe "#activity" do
+    it "returns the activity the csv row represents" do
+      csv_row = double(CSV::Row)
+      allow(csv_row).to receive(:field).with("Activity RODA Identifier")
+      row = described_class.new(csv_row)
+
+      expect(row.activity).to be(@activity)
+    end
+
+    context "when the RODA Identifier cannot be found" do
+      it "returns nil and adds an error" do
+        csv_row = double(CSV::Row)
+        allow(csv_row).to receive(:field).with("Activity RODA Identifier")
+        allow(Activity).to receive(:find_by_roda_identifier).and_return(nil)
+        row = described_class.new(csv_row)
+
+        expect(row.activity).to be_nil
+        expect(row.errors["Activity RODA Identifier"][1]).to include("cannot be found")
+      end
+    end
+  end
+
+  describe "#roda_identifier" do
+    it "returns the identifier string" do
+      csv_row = double(CSV::Row)
+      allow(csv_row).to receive(:field).with("Activity RODA Identifier").and_return("RODA-ID")
+      row = described_class.new(csv_row)
+
+      expect(row.roda_identifier).to eql("RODA-ID")
+    end
+
+    context "when the RODA Identifier is blank" do
+      it "returns blank" do
+        csv_row = double(CSV::Row)
+        allow(csv_row).to receive(:field).with("Activity RODA Identifier").and_return(nil)
+        row = described_class.new(csv_row)
+
+        expect(row.roda_identifier).to eql(:blank)
+      end
+    end
+  end
+
+  describe "#actual_value" do
+    it "returns the value from the csv row" do
```

Comment:
The description of this spec feels incomplete. It's not just returning the value, it's converting it into a BigDecimal.

It feels like there's an implicit context here too, "when the value is numeric".

What happens when the value is zero?

Reply:
👍 Fair point, I'll update the spec.